source/MaterialXCore/Node.cpp

```diff
@@ -102,16 +102,50 @@ Edge Node::getUpstreamEdge(ConstMaterialPtr material, size_t index) const
     return NULL_EDGE;
 }
 
-OutputPtr Node::getNodeDefOutput(const Edge& edge)
+OutputPtr Node::getNodeDefOutput(ElementPtr connectingElement)
 {
-    const ElementPtr connectingElement = edge.getConnectingElement();
-    const PortElementPtr input = connectingElement ? connectingElement->asA<PortElement>() : nullptr;
-    if (input)
+    const string* outputName = nullptr;
+    const PortElementPtr port = connectingElement->asA<PortElement>();
+    if (port)
     {
+        // The connecting element is an input/output port,
+        // so get the name of the output connected to this 
+        // port. If no explicit output is specified this will
+        // return an empty string which is handled below.
+        outputName = &port->getOutputString();
+    }
+    else
+    {
+        const BindInputPtr bindInput = connectingElement->asA<BindInput>();
+        if (bindInput)
+        {
+            // Handle the case where the edge involves a bindinput.
+            const OutputPtr output = bindInput->getConnectedOutput();
```

Comment:
Does the search for an output on a graph vs top level only for bindinputs (which mirrors what's the the "findRenderableElements" utility, or can this occur for other inputs?

Reply:
bindinput are special since they can connect to both graph outputs and top-level outputs. Ordinary node inputs can only connect to other node's outputs.